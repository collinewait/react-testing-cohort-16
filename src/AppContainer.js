import React from 'react';
import App from './App';

export class AppContainer extends React.Component {
    state = {
        value1: null,
        value2: null,
        answer: ''
    }

    add = (value1, value2) => {
        this.setState({answer: (value1 + value2)});
    }

    handleChange = (event) => {
        this.setState({[event.target.name]: parseInt(event.target.value)});
    }    
    
    handleSubmit = (e) => {
        e.preventDefault();
        const { value1, value2 } = this.state;
        this.add(value1, value2)
    }

    render(){
        return (
            <App  
                handleChange={this.handleChange}
                handleSubmit={this.handleSubmit}
                answer={this.state.answer}
            />
        );
    }
}

export default AppContainer;