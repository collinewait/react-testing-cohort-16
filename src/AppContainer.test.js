import React from 'react';
import { shallow, mount } from 'enzyme';

import { AppContainer }  from './AppContainer';

describe('AppContainer', () => {
    let mountAppContainer;
    let instance;

    beforeEach(() => {
        mountAppContainer = shallow(<AppContainer />)
        instance = mountAppContainer.instance();
    });
    
    it('should render without crashing', () => {
        expect(mountAppContainer).toMatchSnapshot()
    });

    it('should update state to the value recieved from the event', () => {
        instance.handleChange({target: {value: '6', name: 'value1'}})
        expect(instance.state.value1).toEqual(6);
    });

    it('should update answer state variable after calling add method', () => {
        instance.add(4, 6)
        expect(instance.state.answer).toEqual(10);
    });

    it('should call add method when handle submit is called', () => {
        const spyAdd = jest.spyOn(instance, 'add');
        instance.handleSubmit({preventDefault: jest.fn()})
        expect(spyAdd).toHaveBeenCalled();
    });

});

// it('should call add method when handle submit is called', () => {
//     const mountAppContainer = mount(<AppContainer />)
//     const spyAppContainer = jest.spyOn(mountAppContainer.instance(), 'handleChange');
//     mountAppContainer.instance().forceUpdate();
//     mountAppContainer.find('#value1').simulate('change');
//     expect(spyAppContainer).toHaveBeenCalledTimes(1);
// });