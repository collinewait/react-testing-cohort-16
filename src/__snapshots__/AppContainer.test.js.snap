// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer should render without crashing 1`] = `
<App
  answer=""
  handleChange={[Function]}
  handleSubmit={[Function]}
/>
`;
